src/datasource.js

```diff
@@ -246,6 +277,40 @@ export class GenericDatasource {
     return graphs;
   }
 
+  /**
+   * Transform data returned by time series query into Grafana annotation format.
+   * @param results
+   * @param options
+   * @returns Array
+   */
+  static transformAnnotationResults(results, timeField, timeEndField, textField) {
+    const annotations = [];
+    results.forEach((result) => {
+      const responseObject = {};
+      responseObject.time = Number(result[timeField]) / 1000000;
+      if (!responseObject.time && result.attributes) {
+        responseObject.time = Number(result.attributes[timeField]) / 1000000;
+      }
+      if (!responseObject.time) {
```

Comment:
It seems like you are trying to avoid some extra processing by returning early if `time` is not defined.  If they have their query correct, `time` should be defined.

From Google, there is no `forEach` continue.  The `return` is returning from the function early, which is slightly hacky.  I think this code is clearer as:
```
if (responseObject.time) {
   annotations.push(responseObject);
}
```
We may do a little more processing getting the `message` field if `time` is not defined, but I think the code is clearer that way.